ampd/src/event_processor.rs

```diff
@@ -491,4 +516,57 @@ mod tests {
         .to_any()
         .unwrap()
     }
+    #[tokio::test]
```

Comment:
```suggestion
    
    #[tokio::test(start_paused = true)]
```
we don't want to actually wait for sleeps etc during the tests. With start_paused, tokio will instead use simulated time. That means sleep etc will increment the internal clock by that time, but the system will not just idle when it's blocking